adbnx_adapter/adbnx_adapter/adbnx_adapter.py

```diff
@@ -304,16 +320,22 @@ def __insert_arangodb_vertex(self, id, v: dict, col: str, key: str, ow: bool):
         :type col: str
         :param key: The _key value of the vertex.
         :type key: str
-        :param ow: If set to True, overwrite existing document with matching _id.
-        :type ow: bool
+        :param vertices: The ArangoDB vertex collection dictionary
+        :type vertices: dict
         """
         self.cntrl.adb_map[id] = {"_id": v["_id"], "collection": col, "key": key}
-        self.db.collection(col).insert(v, overwrite=ow, silent=True)
+        v_col: list = vertices[col]
+        v_col.append(v)
```

Comment:
It looks like the input vertices are all buffered into lists inside of this dictionary and then written all at once? In the case of very large networks that might not be efficient, as it will be one giant network call/insertion per collection (see https://docs.python-arango.com/en/main/_modules/arango/collection.html#Collection.import_bulk). 

Would it make sense to flush the `v_col` list (or `e_col` in the case of edges) if it has reached a certain length? e.g. if there are 1000 elements in the list, insert those 1000 elements and then clear the list. Of course, once it's done with the calls to `__insert_arangodb_vertex` and `__insert_arangodb_edge`, the final write will still need to happen in order to take care of the smaller lists.

The batch size for flushing to ArangoDB could be a parameter? 🤔 

This will definitely increase the complexity of the insertion logic, but has the benefit of smaller network calls & a (somewhat) bounded memory footprint.

Reply:
done! 🚀 

Users can provide a `batch_size` parameter to specify the amount of elements needed before an array flush, and we've removed any batch insertion logic deprecated by ArangoDB.

We're now oriented around smaller network calls to favour a lighter memory footprint